org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/IMavenLauncher.java

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Christoph Läubrich
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *      Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.core.ui.internal;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.core.runtime.CoreException;
+
+
+/**
+ * {@link IMavenLauncher} allows to trigger a maven run
+ */
+public interface IMavenLauncher {
+
+  CompletableFuture<?> runMaven(File basedir, String goals, Properties properties) throws CoreException;
```

Comment:
Couldn't this be added to `AbstractMavenRuntime` ?

Reply:
That's absolutely possible, sad enough I haven't noticed that class before :-\

Reply:
@mickaelistria the runtime can be used to create launches but not to actually perform a launch or do I oversee something?

Reply:
> @mickaelistria the runtime can be used to create launches but not to actually perform a launch or do I oversee something?

The actual launching, i.e. assembling all command line arguments and finally starting the Java-process, is done my JDT and I assume replicating that is in the end just more complicated than just creating a hidden launch as you did.

Reply:
That's why I think I cant just add to AbstractMavenRuntime here... but it would make sense to move the interface there...

Reply:
> he runtime can be used to create launches but not to actually perform a launch or do I oversee something?

I'm not sure I get the difference. Can you please elaborate?

Reply:
> > I'm not sure I get the difference. Can you please elaborate?
> 
> I actually want to execute a maven run (with console) like when you do `Run As > Maven Build` and while the RunPomAction _uses_ the `AbstractMavenRuntime` to build all the JVM parameters, the actual execution happens as a jvm process and not through `AbstractMavenRuntime`

This also adds the 'feature' that the maven installation selected as default in the workspace preferences is used to create the archetype project. So I consider this a plus besides the fact that we want to run it in an extra process.